pallets/msa/src/tests/governance_tests.rs

```diff
@@ -121,3 +121,17 @@ fn propose_to_be_provider_happy_path() {
 		assert!(Msa::is_registered_provider(_new_msa_id));
 	})
 }
+
+#[test]
+fn propose_to_be_provider_long_name_should_fail() {
+	new_test_ext().execute_with(|| {
+		// Create a new MSA account and request that it become a provider
+		let (_new_msa_id, key_pair) = create_account();
+		let proposal_res = Msa::propose_to_be_provider(
+			RuntimeOrigin::signed(key_pair.public().into()),
+			Vec::from("this_is_a_really_long_name_that_should_fail"),
+		);
+
+		assert_noop!(proposal_res, Error::<Test>::ExceedsMaxProviderNameSize);
+	})
```

Comment:
Simple test making sure we reject a proposal creation if naming length does not match set limit